### ISSUE TITLE
WP_User_Query returns objects when 'fields' => 'ID' is queried

### ISSUE THREAD

**Issue by alecgeatches** [10up/ElasticPress#1609]
**Describe the bug**

Adding `ep_integrate` to a user query with `'fields' => 'ID'` returns an array of objects, which is different from the default behavior for returning just the user IDs.

Default behavior:

```php
$user_ids_search = new \WP_User_Query([
    'ep_integrate' => false,
    'fields' => 'ID',
    'search' => 'admin',
]);

$results = $user_ids_search->get_results();

// $results:
//
// array (size=1)
//  0 => string '1'
```

EP behavior:

```php
$user_ids_search = new \WP_User_Query([
    'ep_integrate' => true,
    'fields' => 'ID',
    'search' => 'admin',
]);

$results = $user_ids_search->get_results();

// $results:
//
// array (size=1)
//   0 => 
//    object(stdClass)[2165]
//      public 'ID' => int 1
//      public 'elasticsearch' => boolean true
```

**Steps to Reproduce**
1. Create a `WP_User_Query` with the `'fields' => 'ID'` and `'ep_integrate' => true` parameters.
2. Observe that the results of the query contain an array of `stdClass` objects.

**Expected behavior**
The results of the query contain an array of user IDs, without a wrapper object.

**Environment information**
 - Device: PC
 - OS: Windows 10 Pro
 - Browser and version: N/A
 - Plugins and version: ElasticPress 3.3 / ElasticSearch 6.3.2
 - Theme and version: Custom theme
 - Other installed plugin(s) and version(s): N/A

**Additional context**

The difference isn't hard to work around, but it makes migrating existing user queries tricky. This is especially true for queries in which `'ep_integrate' => true` is enabled automatically, and can break existing code.

The problem seems to come from [`format_hits_as_users` in `User\QueryInteraction`](https://github.com/10up/ElasticPress/blob/develop/includes/classes/Indexable/User/QueryIntegration.php#L114), which wraps the results in objects.

[`Post\QueryIntegration` has some separate handlers based on the field type](https://github.com/10up/ElasticPress/blob/develop/includes/classes/Indexable/Post/QueryIntegration.php#L297), which might be a good alternative to handle this problem for users.

**Comment by alecgeatches** [comment]
It's also worth noting that [`WP_User_Query` defaults to `ID`](https://core.trac.wordpress.org/browser/tags/5.3/src/wp-includes/class-wp-user-query.php#L248) if any non-array value is passed into the query. I've noticed that some legacy code incorrectly uses:

```php
get_users([
    'fields' => 'ids',
    // ...
]);
```

and WordPress will return an array of IDs as the result. I'll fix them for my particular case, but it would be great if ElasticPress had the same behavior.

**Comment by tlovett1** [comment]
@petenelson

**Comment by tlovett1** [comment]
Fixed in #1621

**Comment by alecgeatches** [comment]
@petenelson @tlovett1 Much appreciated. Thank you!
